### PR TITLE
Votiro sentinel solutions

### PR DESCRIPTION
**Pull request by sambhrant-metron**
Change(s):
   - Votiro Solution.

   Reason for Change(s):
   - Integration of Votiro v1.0.0

   Version Updated:
   - Yes

   Testing Completed:
   - Yes
   Checked that the validations are passing and have addressed any issues that are present:
   - Yes

### CODE REVIEW

**Comment by v-mchatla** [comment]
Hi @sambhrant-metron,
Can you please create a package with 2.0.0 version using the V2 version of the [tool](https://github.com/Azure/Azure-Sentinel/tree/master/Tools/Create-Azure-Sentinel-Solution/V2).
Thanks

**Comment by sambhrant-metron** [comment]
hi @v-mchatla,
This package is already created using V2 version tool.
Screenshot from 2022-12-16 10-17-04

Please let me know if any changes are needed.

**Comment by v-mchatla** [comment]
Hi @sambhrant-metron,
Thanks for the confirmation. Please update the version in input file to 2.0.0 and recreate the package.
Please create Data folder and place your input file in that. Please refer this [folder](https://github.com/Azure/Azure-Sentinel/tree/master/Solutions/Bitglass) structure we follow.
Please revert the changes from this [path](https://github.com/Azure/Azure-Sentinel/tree/master/Tools/Create-Azure-Sentinel-Solution/V2/input)
Thanks

**Comment by sambhrant-metron** [comment]
hi @v-mchatla,
I will make the necessary changes.
May close the Pull Request and  create a new PR after making.
Thanks

**Comment by v-mchatla** [comment]
> hi @v-mchatla, I will make the necessary changes. May close the Pull Request and create a new PR after making. Thanks

Hi @sambhrant-metron,
Thanks for the update. 
Thanks

**Comment by sambhrant-metron** [comment]
Hi @v-mchatla,
I have made the changes.
Please let me know in-case of any issue.
Thanks

**Comment by sambhrant-metron** [comment]
@microsoft-github-policy-service agree [company="Votiro"]

**Comment by sambhrant-metron** [comment]
@microsoft-github-policy-service agree company="Votiro"

**Comment by v-mchatla** [comment]
Hi @sambhrant-metron,
Thanks for the updated package. I will review and share the feedback.
Thanks

**Comment by v-mchatla** [comment]
Hi @aprakash13,
Can you please review the analytic rules and share your feedback.
Thanks

**Comment by v-mchatla** [comment]
Hi @sambhrant-metron,
I have reviewed the changes, requesting you to fix the following things.
1. Data File: You can refer this sample [document](https://github.com/Azure/Azure-Sentinel/blob/master/Solutions/SlashNext/Data/Solution_SlashNext.json)
         - Can you please modify the logo path it should be Azure Sentinel master log folder path. It will get displayed once the PR gets merged to master.
         - Author details 'Partner' is generic, can you add your organization or team name so that it will be more specific.
         - Description should give a brief idea of what the solution is and how it helps the users. Can you please add clear description.
 2. Can you please share the sample data in .csv file so that I can dump the data to my workspace and test the parser, analytic rules.
 3. Azure Sentinel instances can be replaced by Microsoft Sentinel.
 4. Please add "dataTypesDependencies": [], in workbook metadata.json file
Thanks

**Comment by sambhrant-metron** [comment]
hi @v-mchatla,
I have made all the changes except for the "sample data" file, because the logs for which the parser is built is not in CSV format, rather its in CEF syslog format. The text formatted file is already part of the Solution.
Please let me know if that works.
Thanks.

**Comment by v-mchatla** [comment]
Hi @sambhrant-metron,
Thanks for addressing the comments. Is it possible for you to share the sample data in json format?
Thanks

**Comment by sambhrant-metron** [comment]
hi @v-mchatla,
Thank you for reviewing the files, but the sample data can not be provided in json format.
Please let me know in case of any other changes,
Thanks.

**Comment by v-mchatla** [comment]
Hi @sambhrant-metron,
Let me check with team if there is any other way that we can test this and get back to you.
Thanks

**Comment by v-mchatla** [comment]
Hi @sambhrant-metron,
Please expect some delay in the process as we are occupied with priority work.
Thanks

**Comment by v-mchatla** [comment]
Hi @sambhrant-metron,
Please expect some delay in the process as we are occupied with priority work.
Thanks

**Comment by v-atulyadav** [comment]
Hi @sambhrant-metron,
Please expect some delay in the process, we will update you ASAP. Thanks.

**Comment by v-mchatla** [comment]
Hi @suhelajjaman,
I'm working with the team on ingesting .txt formatted sample data to Log Analytics for testing. I will keep you posted.
Thanks

**Comment by v-atulyadav** [comment]
Looking into this

**Comment by v-mchatla** [comment]
Hi @sambhrant-metron,
Sorry for the delay in response. I have dumped the data into logAnalytics and verified the parser. I will verify the other content and provide you the update by today EOD.
Thanks

**Comment by v-mchatla** [comment]
Hi @sambhrant-metron,
Sorry for the delay in response. Only Anlytic Rules review is pending as those are pointing to custom tables, I will get back to you once done with the review.
Thanks

**Comment by v-mchatla** [comment]
Hi @sambhrant-metron,
Can you please help me in understanding the below queries.
1. Why are you running the analytic rules on top of Votiro_CL
2. How you are populating the data into the table Votiro_CL.
Thanks

**Comment by v-mchatla** [comment]
> Hi @sambhrant-metron, Can you please help me in understanding the below queries.
> 
> 1. Why are you running the analytic rules on top of Votiro_CL
> 2. How you are populating the data into the table Votiro_CL.
>    Thanks

Hi @sambhrant-metron,
It would be great if you could help me with the above queries.
Thanks

**Comment by sambhrant-metron** [comment]
hi @v-mchatla,
Let me check the DetectionTemplateSchemaValidation issue and fix that.
Then I can help can you with the quries.
Thanks.

**Comment by sambhrant-metron** [comment]
hi @v-mchatla ,
Here are the changes for your queries.
1. Why are you running the analytic rules on top of Votiro_CL
    - It should run on VotiroEvents
    - Added this additional table for validation - but this should be named VotiroEvents table
2. How you are populating the data into the table Votiro_CL
    - Data coming from the Data Connector should populate this table, I have renamed it to VotiroEvents
Hope this helps, and please let me know incase of any query.
Thanks.

**Comment by v-mchatla** [comment]
Hi @sambhrant-metron,
Thanks for the details. I have verified the analytic rules and those are getting created successfully. 
Thanks

**Comment by v-mchatla** [comment]
Hi @devikamehra,
Can you please review the connector and analytic Rules and share your feedback.
Thanks

**Comment by v-mchatla** [comment]
Hi @devikamehra,
Can you please review the connector and analytic Rules and share your feedback.
Thanks

**Comment by v-mchatla** [comment]
Hi @devikamehra,
Can you please review the connector and analytic Rules and share your feedback.
Thanks

File: Solutions/Votiro/Data Connectors/VotiroEvents.json
Comment (devikamehra) [inline Solutions/Votiro/Data Connectors/VotiroEvents.json:126]: Update the value of all these fields.

File: Solutions/Votiro/Parser/VotiroEvents.txt
Comment (devikamehra) [inline Solutions/Votiro/Parser/VotiroEvents.txt:3]: Use parse-kv function to parse all the fields in one go. Fields can be renamed later.
Current implementation would fail if even one attribute is missing in between.

File: Solutions/Votiro/Analytic Rules/VotiroFileBlockedFromConnector.yaml
Comment (devikamehra) [inline Solutions/Votiro/Analytic Rules/VotiroFileBlockedFromConnector.yaml:4]: Description should explain the intend of the rule in plain English instead of explaining each field level detail.

File: Solutions/Votiro/Analytic Rules/VotiroFileBlockedFromConnector.yaml
Comment (devikamehra) [inline Solutions/Votiro/Analytic Rules/VotiroFileBlockedFromConnector.yaml:16]: Is "From" empty if no value is present or it will contain the value null always? 
Also, use =~ instead of ==, if applicable.

File: Solutions/Votiro/Analytic Rules/VotiroFileBlockedFromConnector.yaml
Comment (devikamehra) [inline Solutions/Votiro/Analytic Rules/VotiroFileBlockedFromConnector.yaml:17]: File name and Hash can be entities that can be mapped.
Since we are creating Alert per result, can we summarize the data here to avoid any duplicates in a single execution?

File: Solutions/Votiro/Analytic Rules/VotiroFileBlockedFromConnector.yaml
Comment (devikamehra) [inline Solutions/Votiro/Analytic Rules/VotiroFileBlockedFromConnector.yaml:5]: The analytical rule is missing tactics and techniques.

File: Solutions/Votiro/Analytic Rules/VotiroFileBlockedFromConnector.yaml
Comment (devikamehra) [inline Solutions/Votiro/Analytic Rules/VotiroFileBlockedFromConnector.yaml:17]: Use parse-where function to ensure only required records are captured here.

**Review by devikamehra — CHANGES_REQUESTED** [review]
Ensure all feedbacks in first analytical rule are also considered for second one.

**Comment by v-mchatla** [comment]
Hi @sambhrant-metron -
Can you please address the feedback given by Devika.
Thanks

**Comment by sambhrant-metron** [comment]
Hi, @v-mchatla,
I am currently looking into the changes.
Please request you to give me sometime, as I'm verifying some.
Thanks

**Comment by sambhrant-metron** [comment]
Hi @v-mchatla,
We have done all the changes as per request.
Please review and let me know incase of any query.
Thanks,
Sambhrant.

File: Solutions/Votiro/Data Connectors/VotiroEvents.json
Comment (sambhrant-metron) [reply Solutions/Votiro/Data Connectors/VotiroEvents.json:2]: hi @devikamehra,
We have changed as per requested.

File: Solutions/Votiro/Data Connectors/VotiroEvents.json
Comment (sambhrant-metron) [reply Solutions/Votiro/Data Connectors/VotiroEvents.json]: hi @devikamehra,
We have changed as per requested.

File: Solutions/Votiro/Data Connectors/VotiroEvents.json
Comment (sambhrant-metron) [reply Solutions/Votiro/Data Connectors/VotiroEvents.json]: hi @devikamehra,
We have changed as per requested.

File: Solutions/Votiro/Data Connectors/VotiroEvents.json
Comment (sambhrant-metron) [reply Solutions/Votiro/Data Connectors/VotiroEvents.json]: hi @devikamehra,
We have changed as per requested.

File: Solutions/Votiro/Parser/VotiroEvents.txt
Comment (sambhrant-metron) [reply Solutions/Votiro/Parser/VotiroEvents.txt]: hi @devikamehra,
We have changed as per requested.

File: Solutions/Votiro/Analytic Rules/VotiroFileBlockedFromConnector.yaml
Comment (sambhrant-metron) [reply Solutions/Votiro/Analytic Rules/VotiroFileBlockedFromConnector.yaml]: hi @devikamehra,
We have changed as per requested.

File: Solutions/Votiro/Analytic Rules/VotiroFileBlockedFromConnector.yaml
Comment (sambhrant-metron) [reply Solutions/Votiro/Analytic Rules/VotiroFileBlockedFromConnector.yaml]: hi @devikamehra,
"From" will contain null value always, also we used "=~" instead of "==" where ever it was applicable.

File: Solutions/Votiro/Analytic Rules/VotiroFileBlockedFromConnector.yaml
Comment (sambhrant-metron) [reply Solutions/Votiro/Analytic Rules/VotiroFileBlockedFromConnector.yaml]: hi @devikamehra,
We have changed as per requested.

File: Solutions/Votiro/Analytic Rules/VotiroFileBlockedFromConnector.yaml
Comment (sambhrant-metron) [reply Solutions/Votiro/Analytic Rules/VotiroFileBlockedFromConnector.yaml:5]: hi @devikamehra,
We have changed as per requested.

File: Solutions/Votiro/Analytic Rules/VotiroFileBlockedFromConnector.yaml
Comment (sambhrant-metron) [reply Solutions/Votiro/Analytic Rules/VotiroFileBlockedFromConnector.yaml]: hi @devikamehra,
This is a business rule, we need to get all the coloumns we used in "where" condtion along with "parse-where". We can't use more than one coloumn in parse-where operator. So we have made changes accordingly.
If you know anything else please suggest.
Thanks,
Sambhrant.

**Comment by v-mchatla** [comment]
Hi @devikamehra,
Author has made the requested changes. Can you please review the changes and share your thoughts.
Thanks

File: Solutions/Votiro/Analytic Rules/VotiroFileBlockedFromConnector.yaml
Comment (devikamehra) [inline Solutions/Votiro/Analytic Rules/VotiroFileBlockedFromConnector.yaml:31]: This will fail because you have distinct FileWithConnectorDetails at the end.

Use the function "summarize" to resolve previous comment, not "duplicate".

**Review by devikamehra — CHANGES_REQUESTED** [review]

**Comment by v-mchatla** [comment]
Hi @sambhrant-metron,
Please address the suggested changes.
Thanks

**Comment by v-mchatla** [comment]
Hi @sambhrant-metron,
Sorry for the multiple follow-ups. It would be great if you could address Devika's comments.
Thanks

**Comment by sambhrant-metron** [comment]
Hi @v-mchatla,
We are working on the request and sorry for the delay.
Thanks

File: Solutions/Votiro/Analytic Rules/VotiroFileBlockedFromConnector.yaml
Comment (sambhrant-metron) [reply Solutions/Votiro/Analytic Rules/VotiroFileBlockedFromConnector.yaml]: Hi @devikamehra,
I have used 'summarize' function instead of 'distinct'.
Please review and let me know incase of any query.
Thanks.

**Comment by sambhrant-metron** [comment]
Hi @v-mchatla,
Sorry for the delay.
We have made changes as per the request.
Thanks.

**Comment by v-mchatla** [comment]
Hi @sambhrant-metron,
Thanks for making requested changes.
@devikamehra - can you please review the latest changes and share your feedback.
Thanks

**Comment by saurabh-metron** [comment]
Hi @devikamehra, @v-mchatla,
Could you please help us push this PR forward as this was raised in December 2023 and we would like this to be available for the customers as soon as possible, thanks in advance!

File: Solutions/Votiro/Analytic Rules/VotiroFileBlockedFromConnector.yaml
Comment (devikamehra) [inline Solutions/Votiro/Analytic Rules/VotiroFileBlockedFromConnector.yaml:29]: Use =~ instead of ==

File: Solutions/Votiro/Analytic Rules/VotiroFileBlockedFromConnector.yaml
Comment (devikamehra) [inline Solutions/Votiro/Analytic Rules/VotiroFileBlockedFromConnector.yaml:31]: We have to summarize by all relevant fields. We require SrcFileName, SrcFileSHA256, FileWithConnectorDetails, PolicyName, IncidentURL, SanitizationResult and LogSeverity for mapping.
If we summarize with only one value, we might not get the desired results and the field mapping would essentially fail.

**Review by devikamehra — CHANGES_REQUESTED** [review]

File: Solutions/Votiro/Analytic Rules/VotiroFileBlockedFromConnector.yaml
Comment (sambhrant-metron) [reply Solutions/Votiro/Analytic Rules/VotiroFileBlockedFromConnector.yaml:29]: Hi @devikamehra,
We tried =~ as you suggested but while testing we faced an error showing:
image (1)
So we are not making any changes to this request.
Thanks,
Sambhrant.

File: Solutions/Votiro/Analytic Rules/VotiroFileBlockedFromConnector.yaml
Comment (sambhrant-metron) [reply Solutions/Votiro/Analytic Rules/VotiroFileBlockedFromConnector.yaml]: Hi @devikamehra,
We have added all the required fields.
Please let us know incase of any query.
Thanks,
Sambhrant.

**Comment by v-mchatla** [comment]
Hi @sambhrant-metron,
Apologies for the delay. Please request for review once you are done with the requested changes, so that reviewer can start reviewing the changes.
@devikamehra - Can you please review and provide your approval.
Thanks

**Comment by sambhrant-metron** [comment]
Hi @v-mchatla, @devikamehra,
Please review the changes.
Let me know incase of any query.
Thanks,
Sambhrant.

**Comment by v-atulyadav** [comment]
Hi @sambhrant-metron,
Please address the suggested changes.
Thanks

File: Solutions/Votiro/Analytic Rules/VotiroFileBlockedInEmail.yaml
Comment (sambhrant-metron) [reply Solutions/Votiro/Analytic Rules/VotiroFileBlockedInEmail.yaml:27]: hi @devikamehra,
We cannot use the "has" function with the "parse-where" operator.
The "parse-where" operator is used to extract fields from the raw data using regular expressions and apply filtering conditions to those fields. The output of the "parse-where" operator is a set of fields that are derived from the raw data.
On the other hand, the "has" function is used to check if a field contains a particular value. It operates on a field that already exists in the data and cannot be used to filter fields that are derived from the raw data using the "parse-where" operator.
If we want to use 'has' function we have to use 'where' instead of 'parse-where'.
Thanks,
Sambhrant.

**Comment by sambhrant-metron** [comment]
Hi @v-atulyadav,
I have commented the query asked by @devikamehra .
Please let me know incase of any issue.
Thanks,
Sambhrant.

**Comment by v-mchatla** [comment]
Hi @devikamehra
Can you please review the changes and provide your approval.
Thanks

**Review by devikamehra — CHANGES_REQUESTED** [review]

**Comment by sambhrant-metron** [comment]
Hi @v-mchatla,
We are looking into these requests. 
Thanks,
Sambhrant.

**Comment by v-atulyadav** [comment]
Hi @sambhrant-metron, please check comments from @devikamehra and act accordingly. Thanks

**Comment by sambhrant-metron** [comment]
Hi @v-atulyadav,
Thanks for the reminder.
We are looking into these requests.
Thanks,
Sambhrant.

**Comment by v-atulyadav** [comment]
Hi @sambhrant-metron, 
Thank you for acknowledging. We look forward to hearing back from you.

**Comment by v-atulyadav** [comment]
Hi @sambhrant-metron,
We look forward to hearing back from you.
Thanks

**Comment by v-atulyadav** [comment]
Hi @sambhrant-metron,
any update?

**Comment by sambhrant-metron** [comment]
ji @v-atulyadav,
Apologies for the delay in response.
We are still working on it and we shall update them by mid next week.
Thanks,
Sambhrant

**Comment by v-atulyadav** [comment]
Hi @sambhrant-metron,
Thank you for acknowledging. We look forward to hearing back from you.

**Comment by v-atulyadav** [comment]
Hi @sambhrant-metron, thanks for your acknowledgement.

File: Solutions/Votiro/Workbooks/Votiro Monitoring Dashboard.json
Comment (suraj-metron) [reply Solutions/Votiro/Workbooks/Votiro Monitoring Dashboard.json]: Removed the repeated condition

File: Solutions/Votiro/Workbooks/Votiro Monitoring Dashboard.json
Comment (suraj-metron) [reply Solutions/Votiro/Workbooks/Votiro Monitoring Dashboard.json]: Done

File: Solutions/Votiro/Parser/VotiroEvents.txt
Comment (suraj-metron) [reply Solutions/Votiro/Parser/VotiroEvents.txt]: Added filter for Voiro events as DeviceVendor and DeviceProduct

File: Solutions/Votiro/Analytic Rules/VotiroFileBlockedFromConnector.yaml
Comment (suraj-metron) [reply Solutions/Votiro/Analytic Rules/VotiroFileBlockedFromConnector.yaml]: Hi @devikamehra ,
In this comment, we have been advised to use parse-where instead of "==": https://github.com/Azure/Azure-Sentinel/pull/6905#discussion_r1116723629

Could you please let us know which type of query condition should we use here?
Thanks

**Comment by v-atulyadav** [comment]
Hi @sambhrant-metron, let us know once PR is ready to review. Thanks

**Comment by suraj-metron** [comment]
Hi @v-atulyadav 
We have fixed the review comments.
However, we did had a doubt for one of the review comments and we have reached out to @devikamehra for the same. Waiting for her response.
The comment can be found here: https://github.com/Azure/Azure-Sentinel/pull/6905#discussion_r1158332663

Apart from the above, all the other comments are fixed and the PR is updated.

Thanks

File: Solutions/Votiro/Analytic Rules/VotiroFileBlockedFromConnector.yaml
Comment (devikamehra) [reply Solutions/Votiro/Analytic Rules/VotiroFileBlockedFromConnector.yaml]: @suraj-metron I meant to suggest changes for the last like where you were using strcat_delim.
The where clause was absolutely correct.

Comment (sambhrant-metron) [reply Solutions/Votiro/Analytic Rules/VotiroFileBlockedFromConnector.yaml]: Hi @devikamehra,
Can you please provide any information on how can I change the strcat_delim part to parse_where.
Thanks

**Comment by sambhrant-metron** [comment]
Hi @v-atulyadav,
We have added the a new "threats" in Votiro parser.
Also it would be helpfull if you can please provide any information on how can I change the strcat_delim part to parse_where
as pointed by @devikamehra here https://github.com/Azure/Azure-Sentinel/pull/6905/#discussion_r1202774445,
we would be greatfull to you.
Thanks and Regards,
Sambhrant.

**Comment by devikamehra** [comment]
> Hi @v-atulyadav We have fixed the review comments. However, we did had a doubt for one of the review comments and we have reached out to @devikamehra for the same. Waiting for her response. The comment can be found here: [#6905 (comment)](https://github.com/Azure/Azure-Sentinel/pull/6905#discussion_r1158332663)
> 
> Apart from the above, all the other comments are fixed and the PR is updated.
> 
> Thanks

I looked through the query again and I believe parse-where approach will not work. Please the below query.

VotiroEvents
   | where sanitizationResult has "Blocked" and passwordProtected == "false" and from =~ "null"
   | extend FileWithConnectorDetails = strcat_delim(' ', fileName, 'with the hash', SrcFileSHA256, 'that was sent from connector', connectorName)
   | summarize count() by fileName, SrcFileSHA256, FileWithConnectorDetails, policyName, tostring(incidentURL), sanitizationResult, LogSeverity};Votiro_view

**Comment by v-atulyadav** [comment]
Hi @sambhrant-metron, @devikamehra shared a query with you please act accordingly. Thanks

**Comment by sambhrant-metron** [comment]
Hi @v-atulyadav @devikamehra, 
Thank you so much for providing us the relevant query.
We will add it and update the PR as soon as possible.
Thanks and regards,
Sambhrant.

File: Solutions/Votiro/Analytic Rules/VotiroFileBlockedInEmail.yaml
Comment (devikamehra) [inline Solutions/Votiro/Analytic Rules/VotiroFileBlockedInEmail.yaml:26]: Same logic is used in both the queries. What is the difference here.
Also, there is no entity mapping present here which is ideal to be used in Analytical Rule.

**Review by devikamehra — CHANGES_REQUESTED** [review]

File: Solutions/Votiro/Analytic Rules/VotiroFileBlockedInEmail.yaml
Comment (sambhrant-metron) [reply Solutions/Votiro/Analytic Rules/VotiroFileBlockedInEmail.yaml:26]: Hi @devikamehra ,
The functionality of both is similar, except for the fields after "extend" as the data retrieval looks different.

As for the "entity mapping" part we are looking into it.
Thanks,
Sambhrant

**Comment by v-mchatla** [comment]
Hi @sambhrant-metron 
Thanks for making requested changes, I will get it reviewed and share the feedback.
Thanks

**Comment by sambhrant-metron** [comment]
Hi @v-mchatla ,
Please expect a delay as all the required review checks have not passed.
I will ping you as soon as all the checks have been passed.
Thanks

**Comment by sambhrant-metron** [comment]
Hi @devikamehra ,
Please review the latest changes after adding the entity mapping
Please revert incase of any issues.
Thanks and Regards,
Sambhrant.

**Review by rahul0216 — COMMENTED** [review]
Thanks @sambhrant-metron for the updates. Rules and Parsers look good. I have one query though,
Both the rules will trigger for same condition with almost same results. How having 2 rules for same condition is beneficial?
Also, aggregationKind is AlertPerResult, so if there are 10 results, 10 Incidents will be created. Hope you understand this.

**Comment by rahul0216** [comment]
@v-mchatla I have reviewed Parser and Analytic Rule. They look good. Please review/get reviewed other contents.

**Comment by v-atulyadav** [comment]
Hi @sambhrant-metron, please check above comments from @v-mchatla and act accordingly. Thanks

**Comment by sambhrant-metron** [comment]
hi @v-atulyadav,
I am working on the changes and commit the changes as soon as possible.
Thanks.

**Comment by v-atulyadav** [comment]
Noted @sambhrant-metron, thanks.

**Comment by v-mchatla** [comment]
Hi @sambhrant-metron 
I hope you are doing good. May I know the expected date for publishing the requested changes.
Thanks

**Comment by sambhrant-metron** [comment]
Hi @v-mchatla , 
We have completed all the requested changes and updates. Following are the changes we have done:

- Changed Connectivity criteria query in data connector checking for last 30days data to 3 days.
- Workbook query for Incoming Traffic(Data Source), Added where condition before the summarize operation
- Workbook query for Scanned Files(Threats), Added where condition before the summarize operation
- Repackaged the solution with Version 3.0.0

Please let us know incase of any issues.
Thanks and Regards,
Sambhrant.

**Comment by sambhrant-metron** [comment]
Hi @v-mchatla ,
I have setup a call today at 2:30 pm IST, if you have issues with the time schedule please suggest a new time for the meeting.
Thanks
Sambhrant.

**Comment by v-mchatla** [comment]
Hi @sambhrant-metron 
Thanks for your time over call.
As discussed, We are going to merge this PR and applying below changes on top of your changes.
1.	Workbook metadata changes
2.	Parser file extension change from .txt to yaml
3.	Release Note
4.	Repackaging the solution with 3.0.0
5.	Removing the existing 2.0.0 version from the master and from the 3.0.0 zip
**Note:** Please don’t publish the package until we completed the remaining changes and confirmed, as we can’t publish the same version again.
Thanks

**Review by v-mchatla — APPROVED** [review]

**Review by v-atulyadav — APPROVED** [review]

**Pull request merged**
